soda/cmd/validate/model.go

```diff
@@ -0,0 +1,111 @@
+package validate
+
+import (
+	"fmt"
+	"go/ast"
+	"regexp"
+	"strings"
+)
+
+type model struct {
+	packages map[string]*ast.Package
+	tags     map[string][]string
+	rules    []Checker
+}
+
+type Checker interface {
+	Check(tag string, structName string) []ValidationError
+}
+
+type ValidationError struct {
+	invalidSymbols string
```

Comment:
Why can't we export these fields? It would help people who want to access this programmatically.